### PR TITLE
enable_bf16_vec_cat

### PR DESCRIPTION
**Pull request by zhuhaozhe**
enable_bf16_vec_cat

### CODE REVIEW

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -204,7 +204,8 @@ Tensor & _cat_out_cpu(Tensor& result, TensorList tensors, int64_t dim) {
   allContiguous = allContiguous && result.is_contiguous(first_tensor_mem_format);
   bool use_serial_kernel = result.numel() < at::internal::GRAIN_SIZE || at::get_num_threads() == 1;
   ScalarType dtype = notSkippedTensor.scalar_type();
-  if (use_serial_kernel && allContiguous && no_type_promotion && (dtype == ScalarType::Double || dtype == ScalarType::Float)) {
+  bool serial_dtype = (dtype == ScalarType::Double || dtype == ScalarType::Float || dtype == ScalarType::BFloat16);
```
Comment (heitorschueroff) [inline aten/src/ATen/native/TensorShape.cpp:207]: nice refactor

**Review by heitorschueroff — APPROVED** [review]
LGTM! Thanks for this contribution. @mruberry would you like to take a look and make sure I can land this?

**Comment by mruberry** [comment]
A few questions:

- this seems like 2 changes, not one? should we separate this into 2 PRs?
- do you have performance numbers of program snippets that compare the before and after this PR
- are these code paths tested for bfloat16 in the current test suite?

**Comment by zhuhaozhe** [comment]
@heitorschueroff , @mruberry . Thanks for your reivew. I have separate this into 2 PRs 
https://github.com/pytorch/pytorch/pull/54674
https://github.com/pytorch/pytorch/pull/54671

And for ut, I just grep test_copy and test_cat from pytorch/test forder and pass some related ones. Should CI contains these tests? Since pytorch already support bf16's copy and cat, I think the ut should already cover these situation. Please kindly correct me if I am wrong, thanks~

**Comment by heitorschueroff** [comment]
> #54671

Yes, there's already a test for cat and copy that tests all dtypes. These PR only add support for vectorized operations, these should already be covered as long as the existing tests are large enough to be vectorized.

**Comment by heitorschueroff** [comment]
Closing this PR in favor of https://github.com/pytorch/pytorch/pull/54674 and https://github.com/pytorch/pytorch/pull/54671

**Pull request closed without merging**